packages/core/src/router.ts

```diff
@@ -484,7 +496,7 @@ export class Router {
       Promise.resolve(this.resolveComponent(page.component)).then((component) => {
         if (visitId === this.visitId) {
           this.page = page
-          this.swapComponent({ component, page, preserveState: false }).then(() => {
+          this.swapComponent({ component, page, preserveState: false, preserveScroll: true }).then(() => {
             this.restoreScrollPositions()
```

Comment:
As far as I can tell, `restoreScrollPositions()` doesn't suffer from this same issue in my testing. I'm not entirely sure why, to be honest.